### ISSUE TITLE
Fix `test` verbs usage of `torch.no_grad`

### ISSUE THREAD

**Issue by drewoldag** [lincc-frameworks/hyrax#662]
The `test` verb attempts to make use of `train_step`, while also setting the torch.no_grac context manager. 

These two things are mutually exclusive because in `train_step` we typicially call `loss.backward()` which expects gradients to have been tracked, and it calls `self.optimizer.step()` which updates the model weights. 

This agreed upon approach is now: 
1) Create a separate model method for `test` (and `validate`)
2) Create a separate method for `infer` as well.
3) Name all of these functions `*_batch`, i.e. `train_batch`, `validate_batch`, `test_batch`, `infer_batch`.
4) Update the existing built in models so that they implement those correctly
5) Update the pytorch_ignite engine machinery so that it calls the correct model methods.
